File: jzoffer 21-40/21.cpp
class Solution {
public:
    bool IsPopOrder(vector<int> pushV,vector<int> popV) {
        int id=0;
        if(pushV.empty() ||popV.empty()) return false;
        
        for(int i=0;i<popV.size();i++){
            while(a.empty() || a.top()!=popV[i]){
                a.push(pushV[id]);
                id++;
                if(id>popV.size()){
                    return false;
                }
            }
            a.pop();
        }
        if(a.empty()) return true;
        else {return false;}
    }
private:
    stack<int> a;
    
    
};

File: jzoffer 21-40/21.js
function IsPopOrder(pushV, popV)
{
    // write code here
    var a=[];
    var id=0;
    for(let i=0;i<popV.length;i++){
        while(a.length==0 || a[a.length-1]!=popV[i]){
          a.push(pushV[id]);
            id++;
        if(id>popV.length) return false;
        }
        a.pop();
    }
    if(a.length==0) return true;
    else return false;
}